spec/lib/salus/scanners/bandit_spec.rb

```diff
@@ -0,0 +1,437 @@
+        expect(repo.setup_cfg_present?).to eq(true)
+        expect(scanner.should_run?).to eq(true)
+      end
+    end
+  end
+
+  describe '#run' do
+    before { scanner.run }
+    let(:scanner) { Salus::Scanners::Bandit.new(repository: repo, config: {}) }
+
+    context 'non-bandit project' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_empty_code") }
+
+      it 'should record 0 line of code scanned if no code' do
+        expect(scanner.should_run?).to eq(true)
+        expect(scanner.report.passed?).to eq(false)
+        errs = scanner.report.to_h.fetch(:errors)
+        expect(errs.size).to eq(1)
+        expect(errs[0][:message]).to eq('0 lines of code were scanned')
+      end
+    end
+
+    context 'python project with insecure code' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_with_insecure_code_practices") }
+
+      it 'should record failure and record the STDOUT from bandit' do
+        expect(scanner.report.passed?).to eq(false)
+
+        info = scanner.report.to_h.fetch(:info)
+        logs = scanner.report.to_h.fetch(:logs)
+        expect(info[:stdout]).not_to be_nil
+        expect(info[:stdout]).not_to be_empty
+        msg = 'Consider possible security implications associated with cPickle module.'
+        expect(logs).to include(msg)
+      end
+    end
+
+    context 'python project with insecure code in a nested folder' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_with_insecure_code_practices_r") }
+
+      it 'should record failure and record the STDOUT from gosec' do
+        expect(scanner.report.passed?).to eq(false)
+
+        info = scanner.report.to_h.fetch(:info)
+        logs = scanner.report.to_h.fetch(:logs)
+        expect(info[:stdout]).not_to be_nil
+        expect(info[:stdout]).not_to be_empty
+        msg = 'Consider possible security implications associated with cPickle module.'
+        expect(logs).to include(msg)
+      end
+    end
+
+    context 'python project with no known vulnerabilities' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_no_vulns") }
+
+      it 'should report a passing scan' do
+        expect(scanner.report.passed?).to eq(true)
+      end
+    end
+
+    context 'python project with unknown skip' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_with_bandit_errors") }
+
+      it 'should report a failing scan' do
+        expect(scanner.report.passed?).to eq(false)
+
+        info = scanner.report.to_h.fetch(:info)
+        expect(info[:stderr]).to include("ERROR\tUnknown test found in profile: B9999999")
+      end
+    end
+
+    context 'different config options' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_vulns") }
+
+      it 'aggregate option should work' do
+        repo = Salus::Repo.new("#{py_dir}/python_project_vulns2")
+
+        # aggregate by file
+        config_file = "#{py_dir}/salus_configs/aggregate_file.yaml"
```

Comment:
Any reason not to skip file parsing and just pass the config options directly to the scanner?

```ruby
scanner = Salus::Scanners::Bandit.new(repository: repo, config: {"aggregate" => "file"})
```